management_app/views/course.py

```diff
@@ -59,57 +60,44 @@ def offerings():
-    #     units = row[2]            
-    #     course_level = row[3].strip()
-
-    #     # combine_with column may be like "CS 143B" or "CS143B" -> all convert to "CS143B"
-    #     if pd.isna(row[4]):
-    #         combine_with = None
-    #     else:            
-    #         combine_with = str(row[4]).strip().replace(' ', '')
-        
-
-    #     db.execute(
-    #         'INSERT INTO courses (course_title_id, course_title, units, course_level, combine_with)'
-    #         ' VALUES (?, ?, ?, ?, ?)',
-    #         (course_title_id, course_title, units, course_level, combine_with)
-    #     )
-    #     db.commit()
-
-
-    # TODO: show "combine_with" column in the front end so that admin will remember to edit it if he needs to
-    # TODO: check: the "unit" column of the courses file only has single values (in the unit column, I won’t get values like 2-4 or 4+2), if not: generate warning and reject add/edit
-
-    db = get_db()
+    db = get_db()    
     courses = db.execute('SELECT * FROM courses').fetchall()
-
     return render_template('courses/catalog.html', courses=courses)
 
 
 @courses.route('/data-templates/<filename>', methods=['GET'])
 @login_required
 def download_template(filename):
-
-    # TODO: do template Prefill: 
-    # Schedule_teaching template prepopulate: year, quarter, usename, ucinetid, 
-        # course_type： Lec, course_sec: A (each prodessor 3 rows: fall, winter, spring)
-    # courses template: no need to prepopulate
+    db = get_db()
+    res = db.execute('SELECT COUNT(*) FROM scheduled_teaching').fetchone()
+    # Prepopulate schedule_teaching template after the first time upload
+    # Should include: year, quarter, ucinetid, course_sec
+    # Each prodessor has 3 rows: fall, winter, spring
+    if res != 0:
+        # get the latest academic year from scheduled_teaching
+        year = db.execute(
+            'SELECT DISTINCT year FROM scheduled_teaching ORDER BY year DESC LIMIT 1'
+        ).fetchone()['year']
+
+        ucinetids = get_ucinetid()
+        rows_data = []
+        for ucinetid in ucinetids:
+            rows_data.append([year, "Fall", ucinetid, "", "A", "", ""])
+            rows_data.append([year+1, "Winter", ucinetid, "", "A", "", ""])
+            rows_data.append([year+1, "Spring", ucinetid, "", "A", "", ""])
+
+        df1 = pd.DataFrame([["Scheduled_teaching", "quarter", "1 (Fall), 2 (Winter), 3 (Spring), 4 (Summer)"], ["", "course_title_id", "CS143B"], ["", "course_sec", "A"], ["", "num_of_enrollment", "180"], ["", "offload_or_recall_flag", "1 for \"offload / recall\" or 0"]], 
+            columns=[" ", "column name", "column values"])
+
+        df2 = pd.DataFrame(rows_data,
+            columns=["year", "quarter", "user_UCINetID",	"course_title_id", "course_sec", "num_of_enrollment", "offload_or_recall_flag"])    
+
+        with pd.ExcelWriter(os.path.join(BASE_DIR, DOWNLOAD_FOLDER, 'scheduled_teaching.xlsx'), mode='w') as writer:  
```

Comment:
Because we store the template file on the server side, if we pre-populate the data in this way, the template will be re-write and sent to the client. Therefore, the server will store the modified template.

Just in case developers send the modified file with sensitive data to GitHub by mistake, do you think we should not store the modified file after sending the file to the client? In this way, we still can have the original empty download template in the server.

Reply:
I think that would be fine if we don't push the modified excel file to Github, so this pre-populated way is okay.

Reply:
Added "management_app/static/data_templates/*" into .gitignore file

Reply:
Just found the template has been pushed after being modified. I'll fix it in another PR.